app/facades/order/order_facade.rb

```diff
@@ -0,0 +1,31 @@
+class Order
+  class OrderFacade
+    def initialize; end
+
+    def orders
+      @orders ||= menu.orders
+    end
+
+    def meals(order)
+      order.meals.map { |m| "#{m.course}: #{m.name}" }
+    end
+
+    def total_cost
+      ordered_meals.map(&method(:current_price)).reduce(:+)
+    end
+
+    private
+
+    def menu
+      @menu ||= Menu.last
+    end
+
+    def current_price(meal)
+      menu.menu_items.find_by(meal_id: meal.id).price
+    end
+
+    def ordered_meals
+      menu.orders.flat_map(&:meals)
```

Comment:
What about using `has_many :assoc_name, through:`  + `distinct` scope to avoid using `flat_map`?